core/drivers/pl011.c

```diff
@@ -131,10 +133,12 @@ static int pl011_getchar(struct serial_chip *chip)
 static void pl011_putc(struct serial_chip *chip, int ch)
 {
 	vaddr_t base = chip_to_base(chip);
+	int count = 0;
 
 	/* Wait until there is space in the FIFO */
 	while (read32(base + UART_FR) & UART_FR_TXFF)
-		;
+		if (count++ == 1000)
```

Comment:
It looks like we could loose characters during intensive printing. Wouldn't it be better to check if uart is enabled (`UART_CR_UARTEN`) or something?

Reply:
Yeah, let's hope we can get read the enabled bit even if the UART clock is turned off. I'll try to do that.